src/libraries/System.Net.Http/src/System/Net/Http/Headers/AltSvcHeaderValue.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Net.Http.Headers
+{
+    public sealed class AltSvcHeaderValue
+    {
+        public static AltSvcHeaderValue Clear { get; } = new AltSvcHeaderValue("clear", null, 0, TimeSpan.Zero);
+
+        public string AlpnProtocolName { get; }
+
+        /// <summary>
+        /// The name of the host serving this alternate service.
+        /// If null, the alternate service is on the same host this header was received from.
+        /// </summary>
+        public string Host { get; }
+        public int Port { get; }
+
+        /// <summary>
+        /// The time span this alternate service is valid for.
+        /// If not specified by the header, defaults to 24 hours.
+        /// </summary>
+        public TimeSpan MaxAge { get; }
```

Comment:
`TimeSpan.Zero` is ma=0.

I don't think we need a nullable here. This value has a default if not present.

Reply:
I don't see a default defined in the spec.

In general, it's important to be able to clearly represent exactly what was transmitted in the header. Even more so if this type is used by the server to create response headers. Without MaxAge being nullable, it's ambiguous if the ma parameter should be included in the header when the value is zero.

Reply:
[Section 3.1](https://tools.ietf.org/html/rfc7838#section-3.1):

> When an alternative service is advertised using Alt-Svc, it is considered fresh for 24 hours from generation of the message.  This can be modified with the "ma" (max-age) parameter.

Regardless, we can table this for now -- I am going to make this header type private because I don't believe it can be meaningfully used by clients. We can handle making it public in a future issue if the desire is there.